src/dbnode/persist/fs/seek_manager.go

```diff
@@ -270,12 +271,18 @@ func (m *seekerManager) CacheShardIndices(shards []uint32) error {
 
 		wg.Add(1)
 		m.cacheShardIndicesWorkers.Go(func() {
+			defer wg.Done()
 			if err := m.openAnyUnopenSeekersFn(byTime); err != nil {
+				// This is expected is the cleanup manager may have
+				// removed out of retention index filesets.
+				if xerrors.Is(err, syscall.ENOENT) {
```

Comment:
I think it is a bit risky to let the presence of this error (which we have seen happening, and expect to happen from time to time) within a multi-error mask the presence of any other errors there.

I think intercepting this error at a lower level would be more bullet proof. In fact, `openAnyUnopenSeekers` already tries to handle that:
```go
	for t := start; !t.After(end); t = t.Add(blockSize) {
		byTime.Lock()
		_, err := m.getOrOpenSeekersWithLock(xtime.ToUnixNano(t), byTime)
		byTime.Unlock()
		if err != nil && err != errSeekerManagerFileSetNotFound {
			multiErr = multiErr.Add(err)
		}
	}
```

The problem, I think, only happens because this `errSeekerManagerFileSetNotFound` can only be returned from a check that the fileset exists, which happens before actually trying to open it a bit later, which I think is the actual window for the race:
```go
func (m *seekerManager) newOpenSeeker(
	shard uint32,
	blockStart time.Time,
	volume int,
) (DataFileSetSeeker, error) {
	exists, err := DataFileSetExists(
		m.filePathPrefix, m.namespace, shard, blockStart, volume)
	if err != nil {
		return nil, err
	}
	if !exists {
		return nil, errSeekerManagerFileSetNotFound
	}

	(...)

	err = seeker.Open(m.namespace, shard, blockStart, volume, resources)

	(...)

}
```

Perhaps, adding error check after `seeker.Open` to emit `errSeekerManagerFileSetNotFound` would be the optimal solution here?